Add NLS messages for error path scenarios.

aguibert opened the pull request:
Also add automated testing of most error messages, and high-level comments
to JSF Container java code.

aguibert commented:
#build

sishida commented on dev/com.ibm.ws.jsf.shared/resources/com/ibm/ws/jsf/shared/resources/messages.nlsprops:
```diff
@@ -16,6 +16,8 @@
 #NLS_MESSAGEFORMAT_VAR
 # -------------------------------------------------------------------------------------------------
 
+# JSFG0100-JSFG0199 reserved for jsfContainer-2.X features
```
Could this message be a complete sentence? Maybe add a verb after the subject and a period at the end?
JSFG0100-JSFG0199 is reserved for jsfContainer-2.X features.

aguibert replied:
this is just a comment in the file, not an actual message

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+#NLS_MESSAGEFORMAT_VAR
+# -------------------------------------------------------------------------------------------------
+
+# JSFG0100-JSFG0199 reserved for jsfContainer-2.X features
```
Suggested change: JSFG0100-JSFG0199 is reserved for jsfContainer-2.X features.

aguibert replied:
this is just a comment in the file, not an actual message

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# JSFG0100-JSFG0199 reserved for jsfContainer-2.X features
+
+# 0: JSF provider type (will be either MOJARRA or MYFACES)
```
Suggested change: The JSF provider type is either MOJARRA or MYFACES.

aguibert replied:
this is a comment in the file (any line that starts with the character `#` is a comment)

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# 0: JSF provider type (will be either MOJARRA or MYFACES)
+# 1: application name
+jsf.container.init=JSFG0100I: Initialized Liberty JSF integrations for JSF provider {0} on application {1}
```
What is doing the initialization? I would say, "_Subject_ initialized Liberty JSF integrations for JSF provider {0} on application {1}."

aguibert replied:
Liberty is doing the initialization, so I think saying "Liberty initialized Liberty JSF integrations ..." would be odd.  
How about:
> Liberty initialized JSF integrations ...

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: JSF provider type (will be either MOJARRA or MYFACES)
+# 1: application name
+jsf.container.init=JSFG0100I: Initialized Liberty JSF integrations for JSF provider {0} on application {1}
+jsf.container.init.explanation=Liberty JSF integrations have been initialized for the indicated application, which will allow CDI and JSF integrations to function.
```
Could we make this message present tense instead?
Liberty JSF integrations are initialized for the indicated application, which allows CDI and JSF integrations to function.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 1: application name
+jsf.container.init=JSFG0100I: Initialized Liberty JSF integrations for JSF provider {0} on application {1}
+jsf.container.init.explanation=Liberty JSF integrations have been initialized for the indicated application, which will allow CDI and JSF integrations to function.
+jsf.container.init.useraction=Informational message.  No user action needed.
```
There are two spaces between these sentences. There only needs to be one. :)

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# 0: application name
+# 1: required ApplicationFactory implementations
+jsf.container.no.jsf.impl=JSFG101E: No JSF implementations found in application {0}. A JSF implementation containing one of the following javax.faces.application.ApplicationFactory implementations must be available to the application: {1}
```
Suggested changes: No JSF implementations were found in application {0}. A JSF implementation that contains one of the following javax.faces.application.ApplicationFactory implementations must be available to the application {1}.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: application name
+# 1: required ApplicationFactory implementations
+jsf.container.no.jsf.impl=JSFG101E: No JSF implementations found in application {0}. A JSF implementation containing one of the following javax.faces.application.ApplicationFactory implementations must be available to the application: {1}
+jsf.container.no.jsf.impl.explanation=Liberty JSF integrations are only supported for a specific set of third-party JSF implementations.  One of the listed ApplicationFactory implementations must be available to the indicated application.
```
Use only one space after a sentence ends.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 1: required ApplicationFactory implementations
+jsf.container.no.jsf.impl=JSFG101E: No JSF implementations found in application {0}. A JSF implementation containing one of the following javax.faces.application.ApplicationFactory implementations must be available to the application: {1}
+jsf.container.no.jsf.impl.explanation=Liberty JSF integrations are only supported for a specific set of third-party JSF implementations.  One of the listed ApplicationFactory implementations must be available to the indicated application.
+jsf.container.no.jsf.impl.useraction=Include either the Apache MyFaces or the Mojarra JSF implementation inside of the indicated application.  Alternatively, if you would like to use the default JSF implementation provided by Liberty (Apache MyFaces), use the jsf-2.2 feature instead of jsfContainer-2.2.
```
Use only one space after a sentence ends. Here are my suggested changes:
Include either the Apache MyFaces or the Mojarra JSF implementation inside of the indicated application. Alternatively, if you would like to use the default JSF implementation that is provided by Liberty, Apache MyFaces, use the jsf-2.2 feature instead of jsfContainer-2.2.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+jsf.container.no.jsf.impl.useraction=Include either the Apache MyFaces or the Mojarra JSF implementation inside of the indicated application.  Alternatively, if you would like to use the default JSF implementation provided by Liberty (Apache MyFaces), use the jsf-2.2 feature instead of jsfContainer-2.2.
+
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Unable to obtain application name from JSF application {0}
```
Can we make this a complete sentence?
The _subject_ is unable to obtain the application name from the JSF application {0}.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Unable to obtain application name from JSF application {0}
+jsf.container.no.app.name.explanation=Registration of an application with JSF Container requires the application name to be available by performing a JNDI lookup of 'java:app/AppName'.
```
Maybe add an article before "JSF Container"?
Registration of an application with a JSF Container requires the application name to be available by performing a JNDI lookup of 'java:app/AppName'.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Unable to obtain application name from JSF application {0}
+jsf.container.no.app.name.explanation=Registration of an application with JSF Container requires the application name to be available by performing a JNDI lookup of 'java:app/AppName'.
+jsf.container.no.app.name.useraction=Verify that the jndi-1.0 feature is enabled, and ensure performing a JNDI lookup such as InitialContext.doLookup("java:app/AppName") works from within the application.
```
Suggested changes:
Verify that the jndi-1.0 feature is enabled and ensure that performing a JNDI lookup, such as InitialContext.doLookup("java:app/AppName"), works from within the application.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: application name
+# 1: expected JSF spec API version range
+# 2: actual JSF spec API version used in app
+jsf.container.bad.spec.api.version=JSFG0103E: The JSF specification API version available to application {0} should be within the version range {1} but was {2}
```
Add period at the end of the sentence.
Do users need to complete any actions because of this message? If so, could we include those actions?
Suggested change:
The JSF specification API version available to application {0} is {2}, but it needs to be within the version range {1}.
And then maybe add a sentence about what the user can do.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 1: expected JSF spec API version range
+# 2: actual JSF spec API version used in app
+jsf.container.bad.spec.api.version=JSFG0103E: The JSF specification API version available to application {0} should be within the version range {1} but was {2}
+jsf.container.bad.spec.api.version.explanation=The version of jsfContainer enabled must correspond to the same JSF API version packaged within the indicated application.
```
Suggested change:
The version of jsfContainer that is enabled must correspond to the same JSF API version packaged within the indicated application.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 2: actual JSF spec API version used in app
+jsf.container.bad.spec.api.version=JSFG0103E: The JSF specification API version available to application {0} should be within the version range {1} but was {2}
+jsf.container.bad.spec.api.version.explanation=The version of jsfContainer enabled must correspond to the same JSF API version packaged within the indicated application.
+jsf.container.bad.spec.api.version.useraction=Ensure that the JSF API packaged within the indicated application corresponds to the version of jsfContainer enabled.  Liberty determines the version of the JSF API by checking the 'Specification-Version' manifest attribute of the jar that contains a JSF API class.
```
Only one space after end of sentence.
".jar" must include the dot.
Suggested changes:
Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the 'Specification-Version' manifest attribute of the .jar file that contains a JSF API class.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 2: actual JSF spec API version used in app
+jsf.container.bad.impl.version=JSFG0104E: The JSF implementation version available to application {0} should be within the version range {1} but was {2}
+jsf.container.bad.impl.version.explanation=The version of jsfContainer enabled must correspond to the same JSF implementation version packaged within the indicated application.
+jsf.container.bad.impl.version.useraction=Ensure that the JSF implementation packaged within the indicated application corresponds to the version of jsfContainer enabled.  Liberty determines the version of the JSF implementation by checking the 'Specification-Version' manifest attribute of the jar that contains the MyFaces or Mojarra ApplicationFactory class.
```
For lines 48-50, see comments on lines 41-43.

sishida commented:
Added ID review comments! Thank you!

aguibert commented:
Thanks for reviewing @sishida !  I've addressed all of your review comments.  
Also made some small code updates in test so I'm running another #build

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+jsf.container.no.jsf.impl.useraction=Include either the Apache MyFaces or the Mojarra JSF implementation inside of the indicated application. Alternatively, if you would like to use the default JSF implementation that is provided by Liberty, Apache MyFaces, use the jsf-2.2 feature instead of jsfContainer-2.2.
+
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Liberty is unable to obtain the application name from the JSF application {0}
```
Please add a period at the end of the sentence.

sishida replied:
Maybe something like this?
Liberty is unable to obtain the JSF application {0} application name.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# 0: application name
+# 1: required ApplicationFactory implementations
+jsf.container.no.jsf.impl=JSFG101E: No JSF implementations were found in application {0}. A JSF implementation that contains one of the following javax.faces.application.ApplicationFactory implementations must be available to the application: {1}
```
Remove the colon after "application" and add a period at the end.
...available to the application {1}.

sishida replied:
Maybe something like this?
The application {1} must be able to access a JSF implementation that contains one of the following javax.faces.application.ApplicationFactory implementations.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# 0: JSF provider type (will be either MOJARRA or MYFACES)
+# 1: application name
+jsf.container.init=JSFG0100I: Liberty initialized JSF integrations for JSF provider {0} on application {1}
```
Please add a period at the end of this sentence.

sishida replied:
Maybe something like this? I don't really like it, but there's nothing grammatically wrong with it, and I can't think of anything better!
For JSF provider {0} on application {1}, Liberty initialized JSF integrations.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 2: actual JSF spec API version used in app
+jsf.container.bad.impl.version=JSFG0104E: The JSF implementation version available to application {0} is {2}, but it needs to be within the version range {1}. Ensure that the JSF implementation that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF implementation by checking the 'Specification-Version ...
+jsf.container.bad.impl.version.explanation=The version of jsfContainer that is enabled must correspond to the same JSF implementation version packaged within the indicated application.
+jsf.container.bad.impl.version.useraction=Ensure that the JSF implementation that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF implementation by checking the 'Specification-Version' manifest attribute of the .jar file that contains the MyFaces or Mojarra ApplicationFactory class.
```
We can remove the quotation marks around 'Specification-Version'.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: application name
+# 1: expected JSF spec API version range
+# 2: actual JSF spec API version used in app
+jsf.container.bad.impl.version=JSFG0104E: The JSF implementation version available to application {0} is {2}, but it needs to be within the version range {1}. Ensure that the JSF implementation that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF implementation by checking the 'Specification-Version ...
```
We can remove the quotation marks around 'Specification-Version'.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 2: actual JSF spec API version used in app
+jsf.container.bad.spec.api.version=JSFG0103E: The JSF specification API version available to application {0} is {2}, but it needs to be within the version range {1}. Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the 'Specification-Version' manifest attr ...
+jsf.container.bad.spec.api.version.explanation=The version of jsfContainer that is enabled must correspond to the same JSF API version packaged within the indicated application.
+jsf.container.bad.spec.api.version.useraction=Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the 'Specification-Version' manifest attribute of the .jar file that contains a JSF API class.
```
We can remove the quotation marks around 'Specification-Version'.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: application name
+# 1: expected JSF spec API version range
+# 2: actual JSF spec API version used in app
+jsf.container.bad.spec.api.version=JSFG0103E: The JSF specification API version available to application {0} is {2}, but it needs to be within the version range {1}. Ensure that the JSF API that is packaged within the indicated application corresponds to the version of jsfContainer that is enabled. Liberty determines the version of the JSF API by checking the 'Specification-Version' manifest attr ...
```
We can remove the quotation marks around 'Specification-Version'.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Liberty is unable to obtain the application name from the JSF application {0}
+jsf.container.no.app.name.explanation=Registration of an application with the Liberty JSF Container requires the application name to be available by performing a JNDI lookup of 'java:app/AppName'.
```
Can we remove the quotation marks around 'java:app/AppName'?

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Liberty is unable to obtain the application name from the JSF application {0}
+jsf.container.no.app.name.explanation=Registration of an application with the Liberty JSF Container requires the application name to be available by performing a JNDI lookup of 'java:app/AppName'.
+jsf.container.no.app.name.useraction=Verify that the jndi-1.0 feature is enabled and ensure that performing a JNDI lookup, such as InitialContext.doLookup("java:app/AppName"), works from within the application.
```
Can we remove the quotation marks around "java:app/AppName"? It's fine if they're part of the code, and we can't--just making sure that's the case! :)

aguibert replied:
changing to double-quotes which will convey proper code meaning.

sishida commented on dev/com.ibm.ws.jsfContainer.2.2/resources/com/ibm/ws/jsf/container/resources/JSFContainerMessages.nlsprops:
```diff
@@ -0,0 +1,50 @@
+jsf.container.no.jsf.impl.useraction=Include either the Apache MyFaces or the Mojarra JSF implementation inside of the indicated application. Alternatively, if you would like to use the default JSF implementation that is provided by Liberty, Apache MyFaces, use the jsf-2.2 feature instead of jsfContainer-2.2.
+
+# 0: application name
+jsf.container.no.app.name=JSFG0102E: Liberty is unable to obtain the application name from the JSF application {0} using JNDI.
```
I'd suggest "...by using JNDI."
